.github/workflows/linux_unit_tests_with_latest_deps.yml

```diff
@@ -65,14 +65,34 @@ jobs:
           # "!" negates return code. exit nonzero if any of these deps are found
           ! pip freeze | grep -E "xgboost|catboost|lightgbm|plotly|ipywidgets|category_encoders"
           exit $?
-      - if: ${{ !matrix.core_dependencies }}
+      - if: ${{ !matrix.core_dependencies && (matrix.command == 'git-test-automl' || matrix.command == 'git-test-other') }}
+        name: Installing Dependencies and Prophet
+        run: |
+          pip install virtualenv
+          virtualenv test_python -q
+          source test_python/bin/activate
+          pip install cmdstanpy==0.9.68
+          PYTHON_LIB="/home/runner/work/evalml/evalml/test_python/lib/python${{ matrix.python_version }}/site-packages"
+          CMDSTAN_LIB="${PYTHON_LIB}/cmdstan-2.27.0"
+          python "${PYTHON_LIB}/cmdstanpy/install_cmdstan.py" -d "${PYTHON_LIB}"
+          export CMDSTAN="${CMDSTAN_LIB}"
+          export STAN_BACKEND=CMDSTANPY
+          pip install prophet
+          make installdeps
+          make installdeps
+          make installdeps-test
+          pip uninstall pystan -y
+          pip freeze
+      - if: ${{ !matrix.core_dependencies && (matrix.command == 'git-test-modelunderstanding' || matrix.command == 'git-test-dask') }}
```

Comment:
Don't install prophet for non-core dependencies and if the tests are `git-test-modelunderstanding` and `git-test-dask` because they don't need it